Installing package with dependencies fails on Windows

Attempting to install a package with any dependency from repository instead of npm registry (e.g. `"package-deps": "github:steelbrain/package-deps"` instead of `"package-deps": "latest"`) leads to the following error:
```
C:\Users\username\.atom\packages\defold-ide\node_modules\atom-package-deps\lib\view.js:64 [Package-Deps] Unable to install autocomplete-lua, Error: npm ERR! git clone C:\Users\username\.atom\.apm\_git-remotes\git-github-com-dapetcu21-luaparse-git-1df252af C:\Users\username\AppData\Local\Temp\npm-13068-09b790b7\git-cache-15c834e3\5a1fd2868adecb79f6bc57ee1f1880a3ea83ad8b: Cloning into 'C:\Users\username\AppData\Local\Temp\npm-13068-09b790b7\git-cache-15c834e3\5a1fd2868adecb79f6bc57ee1f1880a3ea83ad8b'...
npm ERR! git clone C:\Users\username\.atom\.apm\_git-remotes\git-github-com-dapetcu21-luaparse-git-1df252af C:\Users\username\AppData\Local\Temp\npm-13068-09b790b7\git-cache-15c834e3\5a1fd2868adecb79f6bc57ee1f1880a3ea83ad8b: error: cannot spawn sh: No such file or directory
npm ERR! git clone C:\Users\username\.atom\.apm\_git-remotes\git-github-com-dapetcu21-luaparse-git-1df252af C:\Users\username\AppData\Local\Temp\npm-13068-09b790b7\git-cache-15c834e3\5a1fd2868adecb79f6bc57ee1f1880a3ea83ad8b: fatal: unable to fork
npm ERR! git clone C:\Users\username\.atom\.apm\_git-remotes\git-github-com-dapetcu21-luaparse-git-1df252af C:\Users\username\AppData\Local\Temp\npm-13068-09b790b7\git-cache-15c834e3\5a1fd2868adecb79f6bc57ee1f1880a3ea83ad8b: 
npm ERR! git clone C:\Users\username\.atom\.apm\_git-remotes\git-https-github-com-dapetcu21-luaparse-git-7c30d46c C:\Users\username\AppData\Local\Temp\npm-13068-09b790b7\git-cache-a4a3f540\5a1fd2868adecb79f6bc57ee1f1880a3ea83ad8b: Cloning into 'C:\Users\username\AppData\Local\Temp\npm-13068-09b790b7\git-cache-a4a3f540\5a1fd2868adecb79f6bc57ee1f1880a3ea83ad8b'...
npm ERR! git clone C:\Users\username\.atom\.apm\_git-remotes\git-https-github-com-dapetcu21-luaparse-git-7c30d46c C:\Users\username\AppData\Local\Temp\npm-13068-09b790b7\git-cache-a4a3f540\5a1fd2868adecb79f6bc57ee1f1880a3ea83ad8b: error: cannot spawn sh: No such file or directory
npm ERR! git clone C:\Users\username\.atom\.apm\_git-remotes\git-https-github-com-dapetcu21-luaparse-git-7c30d46c C:\Users\username\AppData\Local\Temp\npm-13068-09b790b7\git-cache-a4a3f540\5a1fd2868adecb79f6bc57ee1f1880a3ea83ad8b: fatal: unable to fork
npm ERR! git clone C:\Users\username\.atom\.apm\_git-remotes\git-https-github-com-dapetcu21-luaparse-git-7c30d46c C:\Users\username\AppData\Local\Temp\npm-13068-09b790b7\git-cache-a4a3f540\5a1fd2868adecb79f6bc57ee1f1880a3ea83ad8b: 
npm WARN addRemoteGit Error: Command failed: git -c core.longpaths=true config --get remote.origin.url
npm WARN addRemoteGit 
npm WARN addRemoteGit     at ChildProcess.exithandler (child_process.js:206:12)
npm WARN addRemoteGit     at emitTwo (events.js:106:13)
npm WARN addRemoteGit     at ChildProcess.emit (events.js:191:7)
npm WARN addRemoteGit     at maybeClose (internal/child_process.js:877:16)
npm WARN addRemoteGit     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
npm WARN addRemoteGit  dapetcu21/luaparse resetting remote C:\Users\username\.atom\.apm\_git-remotes\git-github-com-dapetcu21-luaparse-git-b80768f9 because of error: { Error: Command failed: git -c core.longpaths=true config --get remote.origin.url
npm WARN addRemoteGit 
npm WARN addRemoteGit     at ChildProcess.exithandler (child_process.js:206:12)
npm WARN addRemoteGit     at emitTwo (events.js:106:13)
npm WARN addRemoteGit     at ChildProcess.emit (events.js:191:7)
npm WARN addRemoteGit     at maybeClose (internal/child_process.js:877:16)
npm WARN addRemoteGit     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
npm WARN addRemoteGit   killed: false,
npm WARN addRemoteGit   code: 1,
npm WARN addRemoteGit   signal: null,
npm WARN addRemoteGit   cmd: 'git -c core.longpaths=true config --get remote.origin.url' }
npm ERR! git clone --template=C:\Users\username\.atom\.apm\_git-remotes\_templates --mirror git@github.com:dapetcu21/luaparse.git C:\Users\username\.atom\.apm\_git-remotes\git-github-com-dapetcu21-luaparse-git-b80768f9: Cloning into bare repository 'C:\Users\username\.atom\.apm\_git-remotes\git-github-com-dapetcu21-luaparse-git-b80768f9'...
npm ERR! git clone --template=C:\Users\username\.atom\.apm\_git-remotes\_templates --mirror git@github.com:dapetcu21/luaparse.git C:\Users\username\.atom\.apm\_git-remotes\git-github-com-dapetcu21-luaparse-git-b80768f9: error: cannot spawn ssh: No such file or directory
npm ERR! git clone --template=C:\Users\username\.atom\.apm\_git-remotes\_templates --mirror git@github.com:dapetcu21/luaparse.git C:\Users\username\.atom\.apm\_git-remotes\git-github-com-dapetcu21-luaparse-git-b80768f9: fatal: unable to fork
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Users\\root\\AppData\\Local\\atom\\app-1.16.0\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\root\\AppData\\Local\\atom\\app-1.16.0\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\root\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\root\\.atom\\.apmrc" "install" "C:\\Users\\root\\AppData\\Local\\Temp\\d-117319-14696-7sk09q.yejj7nwmi\\package.tgz" "--runtime=electron" "--target=1.3.13" "--arch=x64" "--global-style" "--production" "--msvs_version=2012"
npm ERR! node v6.9.5
npm ERR! npm  v3.10.5
npm ERR! code 128

npm ERR! Command failed: git -c core.longpaths=true clone --template=C:\Users\username\.atom\.apm\_git-remotes\_templates --mirror git@github.com:dapetcu21/luaparse.git C:\Users\username\.atom\.apm\_git-remotes\git-github-com-dapetcu21-luaparse-git-b80768f9
npm ERR! Cloning into bare repository 'C:\Users\username\.atom\.apm\_git-remotes\git-github-com-dapetcu21-luaparse-git-b80768f9'...
npm ERR! error: cannot spawn ssh: No such file or directory
npm ERR! fatal: unable to fork
npm ERR! 
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\username\AppData\Local\Temp\apm-install-dir-117319-14696-jlnq3m.0x6zpycik9\npm-debug.log
```
Tested on Windows 7, Windows 10 and Linux, using the latest version of `package-deps` and Atom v1.16.0. Error occurs only in Windows.
Example package: [atom-defold-ide](https://github.com/dapetcu21/atom-defold-ide) - it installs package [autocomplete-lua](https://github.com/dapetcu21/atom-autocomplete-lua), which has [dependency](https://github.com/dapetcu21/atom-autocomplete-lua/blob/master/package.json#L51) that installs from repository.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Changelog
 
+## Upcoming
+
+- Remove `sb-exec` in favor of `BufferedProcess` from Atom builtins
+
 ## 4.6.0
 
 - Remove config file usage, configs are now stored in Atom config store

--- a/package.json
+++ b/package.json
@@ -24,7 +24,6 @@
   "homepage": "https://github.com/AtomLinter/package-deps#readme",
   "dependencies": {
     "atom-package-path": "^1.1.0",
-    "sb-exec": "^4.0.0",
     "sb-fs": "^3.0.0",
     "semver": "^5.3.0"
   },

--- a/src/helpers.js
+++ b/src/helpers.js
@@ -10,7 +10,7 @@ let shownStorageInfo = false
 const VALID_TICKS = new Set(['✓', 'done'])
 const VALIDATION_REGEXP = /(?:Installing|Moving) (.*?) to .* (.*)/
 
-export function exec(command: string, parameters: Array<string>): Promise<{ stdout: string, stderr: string }> {
+function exec(command: string, parameters: Array<string>): Promise<{ stdout: string, stderr: string }> {
   return new Promise(function(resolve) {
     const data = { stdout: [], stderr: [] }
     const spawnedProcess = new BufferedProcess({

--- a/src/helpers.js
+++ b/src/helpers.js
@@ -13,7 +13,7 @@ const VALIDATION_REGEXP = /(?:Installing|Moving) (.*?) to .* (.*)/
 export function exec(command: string, parameters: Array<string>): Promise<{ stdout: string, stderr: string }> {
   return new Promise(function(resolve) {
     const data = { stdout: [], stderr: [] }
-    const spawnedProcess = BufferedProcess({
+    const spawnedProcess = new BufferedProcess({
       command,
       args: parameters,
       stdout(chunk) {

--- a/src/helpers.js
+++ b/src/helpers.js
@@ -3,30 +3,50 @@
 import FS from 'sb-fs'
 import Path from 'path'
 import semver from 'semver'
-import { exec } from 'sb-exec'
+import { BufferedProcess } from 'atom'
 import type { Dependency } from './types'
 
 let shownStorageInfo = false
 const VALID_TICKS = new Set(['✓', 'done'])
 const VALIDATION_REGEXP = /(?:Installing|Moving) (.*?) to .* (.*)/
 
+export function exec(command: string, parameters: Array<string>): Promise<{ stdout: string, stderr: string }> {
+  return new Promise(function(resolve) {
+    const data = { stdout: [], stderr: [] }
+    const spawnedProcess = BufferedProcess({
+      command,
+      args: parameters,
+      stdout(chunk) {
+        data.stdout.push(chunk)
+      },
+      stderr(chunk) {
+        data.stderr.push(chunk)
+      },
+      exit() {
+        resolve({ stdout: data.stdout.join(''), stderr: data.stderr.join('') })
+      },
+      autoStart: false,
+    })
+    spawnedProcess.start()
+  })
+}
+
 export function apmInstall(dependencies: Array<Dependency>, progressCallback: ((packageName: string, status: boolean) => void)): Promise<Map<string, Error>> {
   const errors = new Map()
   return Promise.all(dependencies.map(function(dep) {
-    return exec(atom.packages.getApmPath(), ['install', dep.version ? `${dep.url}@${dep.version}` : dep.url, '--production', '--color', 'false'], {
-      stream: 'both',
-      ignoreExitCode: true,
-    }).then(function(output) {
-      const successful = VALIDATION_REGEXP.test(output.stdout) && VALID_TICKS.has(VALIDATION_REGEXP.exec(output.stdout)[2])
-      progressCallback(dep.name, successful)
-      if (!successful) {
-        const error = new Error(`Error installing dependency: ${dep.name}`)
-        error.stack = output.stderr
-        throw error
-      }
-    }).catch(function(error) {
-      errors.set(dep.name, error)
-    })
+    return exec(atom.packages.getApmPath(), ['install', dep.version ? `${dep.url}@${dep.version}` : dep.url, '--production', '--color', 'false'])
+      .then(function(output) {
+        const successful = VALIDATION_REGEXP.test(output.stdout) && VALID_TICKS.has(VALIDATION_REGEXP.exec(output.stdout)[2])
+        progressCallback(dep.name, successful)
+        if (!successful) {
+          const error = new Error(`Error installing dependency: ${dep.name}`)
+          error.stack = output.stderr
+          throw error
+        }
+      })
+      .catch(function(error) {
+        errors.set(dep.name, error)
+      })
   })).then(function() {
     return errors
   })

